src/SAWScript/Crucible/LLVM/X86.hs

```diff
@@ -297,236 +365,279 @@ llvmSignature opts llvmModule nm =
-  when (isJust cond) $ fail "unsupported x86_64 command: crucible_conditional_points_to"
-  ptr <- C.LLVM.unpackMemValue sym (C.LLVM.LLVMPointerRepr $ knownNat @64)
-    =<< resolveSetupVal cc mem env tyenv Map.empty tptr
-  val <- resolveSetupVal cc mem env tyenv Map.empty tval
-  storTy <- C.LLVM.toStorableType =<< typeOfSetupValue cc tyenv nameEnv tval
-  C.LLVM.storeConstRaw sym mem ptr storTy C.LLVM.noAlignment val
+  X86Sim ()
+executePointsTo env tyenv nameEnv (LLVMPointsTo _ cond tptr tval)
+  | isJust cond = throwX86 "unsupported x86_64 command: crucible_conditional_points_to"
+  | otherwise = do
+      sym <- use x86Sym
+      cc <- use x86CrucibleContext
+      mem <- use x86Mem
+      ptr <- liftIO $ C.LLVM.unpackMemValue sym (C.LLVM.LLVMPointerRepr $ knownNat @64)
+        =<< resolveSetupVal cc mem env tyenv Map.empty tptr
+      val <- liftIO $ resolveSetupVal cc mem env tyenv Map.empty tval
+      storTy <- liftIO $ C.LLVM.toStorableType =<< typeOfSetupValue cc tyenv nameEnv tval
+      mem' <- liftIO $ C.LLVM.storeConstRaw sym mem ptr storTy C.LLVM.noAlignment val
+      x86Mem .= mem'
 
 -- | Write each SetupValue passed to crucible_execute_func to the appropriate
 -- x86_64 register from the calling convention.
 setArgs ::
   X86Constraints =>
-  Sym ->
-  LLVMCrucibleContext LLVMArch ->
   Map MS.AllocIndex Ptr {- ^ Associates each AllocIndex with the corresponding allocation -} ->
   Map MS.AllocIndex LLVMAllocSpec {- ^ Associates each AllocIndex with its specification -} ->
   Map MS.AllocIndex C.LLVM.Ident {- ^ Associates each AllocIndex with its name -} ->
-  Mem ->
-  Regs ->
   [MS.SetupValue LLVM] {- ^ Arguments passed to crucible_execute_func -} ->
-  IO Regs
-setArgs sym cc env tyenv nameEnv mem regs args
-  | length args > length argRegs = fail "More arguments than would fit into general-purpose registers"
-  | otherwise = foldlM setRegSetupValue regs $ zip argRegs args
-  where
-    argRegs :: [Register]
-    argRegs = [Macaw.RDI, Macaw.RSI, Macaw.RDX, Macaw.RCX, Macaw.R8, Macaw.R9]
-    setRegSetupValue rs (reg, sval) = typeOfSetupValue cc tyenv nameEnv sval >>= \ty ->
-      case ty of
-        C.LLVM.PtrType _ -> do
-          val <- C.LLVM.unpackMemValue sym (C.LLVM.LLVMPointerRepr $ knownNat @64)
-            =<< resolveSetupVal cc mem env tyenv nameEnv sval
-          setReg reg val rs
-        C.LLVM.IntType 64 -> do
-          val <- C.LLVM.unpackMemValue sym (C.LLVM.LLVMPointerRepr $ knownNat @64)
-            =<< resolveSetupVal cc mem env tyenv nameEnv sval
-          setReg reg val rs
-        C.LLVM.IntType _ -> do
-          C.LLVM.LLVMValInt base off <- resolveSetupVal cc mem env tyenv nameEnv sval
-          case testLeq (incNat $ W4.bvWidth off) (knownNat @64) of
-            Nothing -> fail "Argument bitvector does not fit in a single general-purpose register"
-            Just LeqProof -> do
-              off' <- W4.bvZext sym (knownNat @64) off
+  X86Sim ()
+setArgs env tyenv nameEnv args
```

Comment:
this function is not easy to review, because of the format changes, which is why I don't like `where`